code-challenges.js

```diff
@@ -23,19 +41,54 @@ var middleLetters1 = "hello"
 var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
+const myFunc = (string) => {
+    let theMiddle = string.substr(string.length - 1 >>> 1, (~string.length & 1) + 1);
+    return theMiddle
+}
 
-
+console.log(myFunc(middleLetters1))
 
 // --------------------4) Create a class to get the area of a sphere. Create three spheres with different radi as test cases. Area of a sphere =  4πr^2 (four pi r squared)
 
+class Sphere{
+    constructor(){
+        this.sphere = 0;
+    }
+    
+}
+
+// i had lots of trouble with this one ill be diffenatly be reviewing this during class :(
+
 
 
 
-// --------------------5) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+
+// ---------------5) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 var numbersToAdd1 = [2, 4, 45, 9]
 // Excpected output: [2, 6, 51, 60]
 var numbersToAdd2 = [0, 7, -8, 12]
 // Expected output: [0, 7, -1, 11]
 var numbersToAdd3 = []
 // Expected output: []
+
+
+
+// creating a function that can take in the arrays above
+// declaring a variable  to hold the index data
+
+const accumulate = (array) =>{
+    let temp = 0;
+    let tempArray = []
+    for (let i = 0; i<array.length; i++){
+        temp = array[i] + tempArray
+        tempArray.push(temp)
+    }
+   
+    return tempArray
+}
```

Comment:
i don't think this returns what you're expecting. one return is `[ '2', '42', '452,42', '92,42,452,42' ]`. i think the problem is when you add `temp` to `tempArray`.